source/core/src/test/com/csse3200/game/entities/factories/NPCFactoryTest.java

```diff
@@ -114,8 +128,285 @@ static void setup() {
+     */
+//    @Test
+//    void TestChickenHasAnimation() {
+//        Assertions.assertTrue(friendlyChicken.getComponent(AnimationRenderComponent.class).hasAnimation("wait"),
+//                "Chicken should have idle animation.");
+//    }
+
+    /**
+     * Tests that the friendly chicken is a friendly NPC meaning it won't attack players.
+     */
+    @Test
+    void TestChickenIsFriendly() {
+        Assertions.assertNotNull(friendlyChicken.getComponent(FriendlyNPCAnimationController.class),
+                "Chicken should have a friendly AI controller.");
+    }
+
+    /**
+     * Tests that the friendly chicken is in the correct spot when placed.
+     */
+    @Test
+    void TestChickenSetPosition() {
+        Vector2 pos = new Vector2(0f, 0f);
+        friendlyChicken.setPosition(pos);
+
+        Assertions.assertEquals(pos, friendlyChicken.getPosition());
+    }
+
+
+
+
+    /**
+     * Tests the initialization of a friendly frog by checking its creation, name, type,
+     * and the presence of necessary components.
+     */
+    @Test
+    void testFrogInitialisation() {
+        // Test creation of the friendlyFrog
+        Assertions.assertNotNull(friendlyFrog, "Frog should not be null.");
+
+        // Test that the friendlyFrog has the correct name
+        String name = configs.friendlyFrog.getAnimalName();
+        Assertions.assertEquals("Frog", name);
+
+        // Test that the friendlyFrog is an Entity
+        Assertions.assertEquals(Entity.class, friendlyFrog.getClass());
+
+        // Test that the friendlyFrog has a PhysicsComponent
+        Assertions.assertNotNull(friendlyFrog.getComponent(PhysicsComponent.class));
+
+        // Test that the friendlyFrog has a PhysicsMovementComponent
+        Assertions.assertNotNull(friendlyFrog.getComponent(PhysicsMovementComponent.class));
+
+        // Test that the friendlyFrog has a ColliderComponent
+        Assertions.assertNotNull(friendlyFrog.getComponent(ColliderComponent.class));
+
+        // Test that the friendlyFrog has a ConfigComponent
+        Assertions.assertNotNull(friendlyFrog.getComponent(ConfigComponent.class));
     }
 
+    /**
+     * Tests that the friendly frog has the correct sound path.
+     * //TODO:Change when sound path is updated
+     */
+    @Test
+    void TestFrogHasCorrectSoundPath() {
+        String[] sound = configs.friendlyFrog.getSoundPath();
+        Assertions.assertNotNull(sound);
+        Assertions.assertArrayEquals(new String[]{"sounds/frog.wav"}, sound);
+    }
+
+    /**
+     * Tests that the friendly frog has the correct base hint.
+     */
+    @Test
+    void TestFrogHasCorrectBaseHint() {
+        String[][] baseHint = configs.friendlyFrog.getBaseHint();
+        assertNotNull(baseHint);
+        Assertions.assertArrayEquals(new String[][]{{"Welcome to Animal Kingdom!","I am Fred the Frog."}},
+                baseHint);
+    }
+
+    /**
+     * Tests that the friendly frog has a wait animation.
```

Comment:
Why is this test commented out? Please either make it work with the current game or remove it.